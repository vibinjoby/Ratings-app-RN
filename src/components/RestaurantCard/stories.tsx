import React from 'react'
import { StyleSheet, View } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import RestaurantCard from './component'
import constants from '../../configs/commonConst'

storiesOf('Components', module).add('RestaurantCard', () => (
  <View style={styles.container}>
    <RestaurantCard
      id={'123'}
      title="Joes Pizzeria"
      restaurantImg={{ uri: constants.DUMMY_PIC }}
      ratings={4.4}
      reviewCount={84}
    />
  </View>
))

const styles = StyleSheet.create({
  container: {
    padding: 20,
  },
})
