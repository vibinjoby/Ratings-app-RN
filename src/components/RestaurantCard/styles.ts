import { StyleSheet } from 'react-native'

import Colors from '../../utilities/colors'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {},
  imgBg: {
    borderRadius: 10,
    height: 141,
  },
  starImg: {
    marginRight: 5,
  },
  ratings: {
    ...Typography.RegularFont,
    color: Colors.white,
    fontSize: 14,
  },
  ratingWrapper: {
    alignSelf: 'flex-start',
    paddingHorizontal: 5,
    paddingTop: 2,
    backgroundColor: Colors.appOrange,
    flexDirection: 'row',
    borderRadius: 8,
    margin: 7,
  },
  starContainer: {
    flexDirection: 'row',
    position: 'absolute',
    bottom: -20,
  },
  restaurantTitle: {
    ...Typography.Title1.bold,
    marginTop: 5,
    fontSize: 24,
  },
  reviews: {
    ...Typography.Caption1.regular,
    fontSize: 14,
    marginTop: -3,
    marginLeft: 10,
  },
})
