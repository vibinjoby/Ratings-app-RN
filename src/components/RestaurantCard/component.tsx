import { useNavigation } from '@react-navigation/native'
import React from 'react'
import {
  Image,
  ImageBackground,
  ImageSourcePropType,
  Text,
  TouchableOpacity,
  View,
} from 'react-native'
import routes from '../../navigations/routes'

import Colors from '../../utilities/colors'
import Stars from '../Stars'
import styles from './styles'

export interface RestaurantCardProps {
  testID?: string
  id: string
  title: string
  ratings: number
  restaurantImg: ImageSourcePropType
  reviewCount: number
  onPress?: () => void
}

const RestaurantCard: React.FC<RestaurantCardProps> = ({
  testID,
  id,
  title,
  ratings,
  restaurantImg,
  reviewCount,
  onPress,
}: RestaurantCardProps) => {
  const navigation = useNavigation()

  const RatingOverview = () => (
    <View style={styles.ratingWrapper}>
      <Image style={styles.starImg} source={require('../../assets/star/star.png')} />
      <Text testID="ratings" style={styles.ratings}>
        {ratings}
      </Text>
    </View>
  )

  return (
    <TouchableOpacity
      testID={testID}
      activeOpacity={0.8}
      style={styles.container}
      //@ts-ignore
      onPress={() => (onPress ? onPress() : navigation.navigate(routes.RESTAURANT_DETAILS, { id }))}
    >
      <View>
        <ImageBackground
          testID="imgBG"
          source={restaurantImg}
          resizeMode="stretch"
          style={styles.imgBg}
        >
          <RatingOverview />
        </ImageBackground>
        <Text testID="title" style={styles.restaurantTitle}>
          {title}
        </Text>
        <View style={styles.starContainer}>
          <Stars
            testID="stars"
            selectedColor={Colors.golden}
            selectable={false}
            selectedStars={ratings}
            starHeight={15}
          />
          <Text style={styles.reviews}>{reviewCount} Reviews</Text>
        </View>
      </View>
    </TouchableOpacity>
  )
}

export default RestaurantCard
