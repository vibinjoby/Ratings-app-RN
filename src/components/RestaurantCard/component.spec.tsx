import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'

import RestaurantCard from './component'

describe('RestaurantCard', () => {
  const MOCK_FN = jest.fn()
  const MOCK_TITLE = 'Joes Pizzeria'
  const MOCK_ID = ''
  const MOCK_IMG = require('../../../src/assets/dummy-restaurant-pic.png')
  const MOCK_RATINGS = 4.4

  it('should render', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <RestaurantCard
          id={MOCK_ID}
          title={MOCK_TITLE}
          restaurantImg={MOCK_IMG}
          ratings={MOCK_RATINGS}
          reviewCount={84}
          onPress={MOCK_FN}
        />
      </NavigationContainer>,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render all values and be clickable', () => {
    const { getByTestId } = render(
      <NavigationContainer>
        <RestaurantCard
          testID="container"
          id={MOCK_ID}
          title={MOCK_TITLE}
          restaurantImg={MOCK_IMG}
          ratings={MOCK_RATINGS}
          reviewCount={84}
          onPress={MOCK_FN}
        />
      </NavigationContainer>,
    )
    expect(getByTestId('ratings').props.children).toBe(MOCK_RATINGS)
    expect(getByTestId('title').props.children).toBe(MOCK_TITLE)
    expect(getByTestId('imgBG').props.source).toBe(MOCK_IMG)
    fireEvent.press(getByTestId('container'))
    expect(MOCK_FN).toBeCalled()
  })
})
