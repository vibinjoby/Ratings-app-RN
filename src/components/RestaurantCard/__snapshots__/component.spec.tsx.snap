// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard should render 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "borderRadius": 10,
          "height": 141,
        }
      }
    >
      <Image
        resizeMode="stretch"
        source={
          Object {
            "testUri": "../../../src/assets/dummy-restaurant-pic.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 141,
              "width": undefined,
            },
            undefined,
          ]
        }
        testID="imgBG"
      />
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#FF5823",
            "borderRadius": 8,
            "flexDirection": "row",
            "margin": 7,
            "paddingHorizontal": 5,
            "paddingTop": 2,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/star/star.png",
            }
          }
          style={
            Object {
              "marginRight": 5,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#fff",
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
            }
          }
          testID="ratings"
        >
          4.4
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#181725",
          "fontFamily": "Poppins-Bold",
          "fontSize": 24,
          "lineHeight": 29,
          "marginTop": 5,
        }
      }
      testID="title"
    >
      Joes Pizzeria
    </Text>
    <View
      style={
        Object {
          "bottom": -20,
          "flexDirection": "row",
          "position": "absolute",
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        data={
          Array [
            1,
            2,
            3,
            4,
            5,
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="stars0"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#F5D547",
                      "fontSize": 15,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="font-icon 0"
              >
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginRight": 5,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="stars1"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#F5D547",
                      "fontSize": 15,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="font-icon 1"
              >
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginRight": 5,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="stars2"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#F5D547",
                      "fontSize": 15,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="font-icon 2"
              >
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginRight": 5,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="stars3"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#F5D547",
                      "fontSize": 15,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="font-icon 3"
              >
                
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginRight": 5,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="stars4"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#F5D547",
                      "fontSize": 15,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="font-icon 4"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <Text
        style={
          Object {
            "color": "#7C7C7C",
            "fontFamily": "Poppins-Regular",
            "fontSize": 14,
            "marginLeft": 10,
            "marginTop": -3,
          }
        }
      >
        84
         Reviews
      </Text>
    </View>
  </View>
</View>
`;
